src/wayland/shell/xdg/mod.rs

```diff
@@ -209,7 +271,11 @@ macro_rules! xdg_role {
 }
 
 xdg_role!(
-    ToplevelConfigure,
+    ToplevelState,
+    /// A configure message for toplevel surfaces
+    #[derive(Debug, Clone)]
+    ToplevelConfigure {
+    },
```

Comment:
nit: empty braces?

Reply:
Yeah, the macro i defined expects them cause `XdgPopup` requires additional fields. I thought about changing the macro, but decided against it to make the macro not more complicated. But i agree that does not look so nice. I changed the macro to allow zero or more repetitions. Now it is possible to write some funny looking invocations, but i am not aware of an alternative without duplicating macro code. And as this macro is not exported it should not matter.